user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md

```diff
@@ -0,0 +1,61 @@
+---
+uid: SwarmingRollback
+---
+
+# Partially rolling back Swarming
+
+> [!IMPORTANT]
+> There is currently no good way to disable Swarming and keep everything in the DMS intact. The info below only partially recovers some data. If this is a test system and all data can be lost, a clean reinstall is advised.
+
+When Swarming is enabled, the element config XML files (e.g. `Element.xml`, `ElementData.xml`, `Description.xml`, etc.) are moved from disk to the database. There is currently no procedure to move them back to disk.
+
+However, there are some things you can do partially recover some data.
+
+## Partial rolling back Swarming from a backup
+
+When DataMiner is moving the XML files from disk to database, it will take a backup first. This migration happens during the first DataMiner startup after the Swarming feature is enabled.
+
+The backup is initially stored in the *Recycle Bin* folder, but you can manually copy it over to a different location to have long-term access. It is stored in two .zip files, one for the elements folder and one for the redundancy folder. These .zip files are a snapshot from the moment Swarming was enabled. The files will for instance look like this:
+
+- `2024_11_20 11_03_12_300_ElementFolder_BeforeSwarmingMigration.zip`
+- `2024_11_20 11_03_12_300_RedundancyFolder_BeforeSwarmingMigration.zip`
+
+To restore this backup and roll back Swarming:
+
+1. Shut down all DMAs in the DMS.
+
+1. On every DMA, restore the `C:\Skyline DataMiner\Elements\` and the `C:\Skyline DataMiner\Redundancy\` folder by replacing the folders on the disk with the ones in the zip files.
+
+   This will restore both the elements and redundancy groups to be back as they were at the time of the backup, meaning that any changes or additions since the backup will be lost.
+
+1. [Manually disable Swarming](#manually-disabling-swarming-in-the-config-files).
+
+1. Restart all DMAs.
+
+> [!IMPORTANT]
+> When you restore the backup, any modifications to existing elements or redundancy groups since the backup and any new elements or redundancy groups created since the backup will be lost. Any reference to these elements could become invalid.
+
+## Restoring elements using an export and import
+
+For small systems, an alternative is to use DELT to extract the latest version of the elements. You would then export the elements and reimport them once Swarming is disabled.
+
+1. [Export the elements](xref:Exporting_elements_services_etc_to_a_dmimport_file) you want to save to a .dmimport file.
```

Comment:
Is this something you should do before you enable Swarming, or can you do this when Swarming is already enabled? If it's the latter, perhaps we should rephrase the sentence below so it mentions "the element configuration" instead of "the element XML files"?

Reply:
It's something you can do while Swarming is enabled. Better to refer to element configuration indeed
A dmimport package taken while swarming is enabled can be restored on a system where swarming is not enabled